### PR TITLE
Implement flask web app

### PR DESCRIPTION
**Pull request by marcelvanworkum**
Implemented the basic structure of the react components, hooked Flask with react and got realtime loading working as well.

Also set up a ton of dependencies: bower, npm, scss, gulp, babel.

### CODE REVIEW

File: software/src/app.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+DEBUG = True
+
+@app.route('/')
+def index():
+    return render_template('index.html')
+
+@app.route('/data')
+def data():
+    data = {
+        'foo': randint(0, 9)
```
Comment (jamiesanson) [inline software/src/app.py]: No `/data` redirects yet?

File: software/src/static/scripts/jsx/main.js [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+  }
+});
+
+var refreshTime = 100;
+
+setInterval(function() {
+  $.get("http://127.0.0.1:5000/data", function (response) {
+    var data = JSON.parse(response);
+
+    ReactDOM.render(
```
Comment (jamiesanson) [inline software/src/static/scripts/jsx/main.js]: Not sure on React; Does this just refresh the 'react-hook' element? Or refresh the entire page

**Review by jamiesanson — COMMENTED** [review]
Not sure how to review as haven't got much experience with web dev, but left a few comments regardless

File: software/src/app.py (hunk 1 above)
Comment (marcelvanworkum) [inline software/src/app.py]: ¿Que?

File: software/src/static/scripts/jsx/main.js (hunk 2 above)
Comment (marcelvanworkum) [reply software/src/static/scripts/jsx/main.js]: Yeah so this is the beauty of React. tl;dr, React performs a diff of the "virtual dom" and finds out what needs to be re-rendered, and not the entire page.

Comment (marcelvanworkum) [reply software/src/static/scripts/jsx/main.js]: https://facebook.github.io/react/docs/reconciliation.html

File: software/src/app.py (hunk 1 above)
Comment (jamiesanson) [reply software/src/app.py]: Dw am dumb

File: software/src/static/scripts/jsx/main.js (hunk 2 above)
Comment (jamiesanson) [reply software/src/static/scripts/jsx/main.js]: That's pretty neat. I like

**Comment by jamiesanson** [comment]
Merge when ready, looks tidy

File: software/src/app.py (hunk 1 above)
Comment (marcelvanworkum) [reply software/src/app.py]: This route is just acting as a GET api to grab our data. The actual web app will be a one pager, so no page reloads/routing needed.

File: software/src/static/scripts/jsx/main.js (hunk 2 above)
Comment (marcelvanworkum) [reply software/src/static/scripts/jsx/main.js]: Yeah React is cool

File: software/src/app.py (hunk 1 above)
Comment (jamiesanson) [reply software/src/app.py]: Yeah I saw that further on, just forgot to remove the comment.

**Comment by marcelvanworkum** [comment]
Sweet, one thing I would like to do is have the components split out into separate files. Not sure on the best way about doing this. I will look into this, maybe ask someone at work and do that before I merge.

**Comment by jamiesanson** [comment]
Sounds good to me. Definitely looks like `main.js` would benefit from being split out into separate files

**Comment by marcelvanworkum** [comment]
Closing for now as development of the base station has been put on the back-burner.

**Pull request closed without merging**